app/controllers/spree/subscriptions_controller.rb

```diff
@@ -23,24 +23,46 @@ def update
     end
 
     def cancel
-      respond_to do |format|
+      if @subscription.stripe_subscription_id.present?
         if @subscription.cancel
-          format.json { render json: {
-              subscription_id: @subscription.id,
-              flash: t(".success"),
-              method: Spree::Subscription::ACTION_REPRESENTATIONS[:cancel].upcase
-            }, status: 200
-          }
-          format.html { redirect_to account_path, success: t(".success") }
+          @stripe_subscription = Stripe::Subscription.retrieve(@subscription.stripe_subscription_id)
+          @stripe_subscription.cancel
```

Comment:
what is subscription is in a state where cancel will fail?

Reply:
i have added begin rescue now in case subscription cancellation halt for whatever reason from stripe then it will give error